source/vst/utility/vst2persistence.cpp

```diff
@@ -589,7 +589,7 @@ bool writeVst2State (const Vst2xState& state, Steinberg::IBStream& _stream,
 			    return res.error;
 		    if (!(res = (stream << state.fxVersion)))
 			    return res.error;
-		    int32_t numPrograms = writeChunk ? 1 : static_cast<int32_t> (state.programs.size ());
+		    int32_t numPrograms = writeChunk ? 0 : static_cast<int32_t> (state.programs.size ());
```

Comment:
This might break backwards compatibility for Plugins compiled against the older version of the `tryVst2StateLoad` code. Please revert!